src/redactableMarkdownProcessor.js

```diff
@@ -1,11 +1,52 @@
 const unified = require("unified");
-const html = require("remark-html");
+// const html = require("remark-html");
+const remarkRehype = require("remark-rehype");
+const rehypeRaw = require("rehype-raw");
+const rehypeSanitize = require("rehype-sanitize");
+const rehypeStringify = require("rehype-stringify");
 const parse = require("remark-parse");
 const stringify = require("remark-stringify");
 const plugins = require("@code-dot-org/remark-plugins");
+const defaultSanitizationSchema = require("hast-util-sanitize/lib/github");
 
 const RedactableProcessor = require("./redactableProcessor");
 
+// create custom sanitization schema as per
+// https://github.com/syntax-tree/hast-util-sanitize#schema
+// to support our custom syntaxes
+const schema = Object.assign({}, defaultSanitizationSchema);
+schema.clobber = [];
+
+// We use a _lot_ of image formatting stuff in our
+// instructions, particularly in CSP
+schema.attributes.img.push("height", "width");
+
+// Add support for expandableImages
+schema.tagNames.push("span");
+schema.attributes.span = ["dataUrl", "className"];
+
+// Add support for inline styles (gross)
+// TODO replace all inline styles in our curriculum content with
+// semantically-significant content
+schema.attributes["*"].push("style", "className");
+
+// ClickableText uses data-id on a bold tag.
+schema.attributes["b"] = ["dataId"];
+
+// Add support for Blockly XML
+const blocklyTags = [
+  "block",
+  "functional_input",
+  "mutation",
+  "next",
+  "statement",
+  "title",
+  "field",
+  "value",
+  "xml",
+];
+schema.tagNames = schema.tagNames.concat(blocklyTags);
+
```

Comment:
This is the same schema used in [`SafeMarkdown`](https://github.com/code-dot-org/code-dot-org/blob/8d2908ab6f9cf5c2ce413958f32e3c27d5ac095e/apps/src/templates/SafeMarkdown.jsx#L68) in our main repo.

Reply:
It would be great to have just one source for our custom schema. Here is a ticket to track that work: [P20-858](https://codedotorg.atlassian.net/browse/P20-858)